plot_features.py

```diff
@@ -134,52 +110,39 @@ def divide_poly_segments(points): #{{{
         else:
             endIndex = inPoints.shape[0]
         outPoints = np.append(outPoints, inPoints[segIndex+1:endIndex,:], axis=0)
-        
+
     return outPoints #}}}
 
 def plot_poly(mapInfo, points, color, filled=True): #{{{
 
     points = divide_poly_segments(points)
-    
+    #points = np.asarray(points)
+
+    refProjection = cartopy.crs.PlateCarree()
+
     for mapIndex in range(len(mapInfo)):
-        (mapType, map, plotFileName, fig, offsets, supportsFill) = mapInfo[mapIndex]
-        plt.figure(fig.number) 
-        for offset in offsets:
-            lon = points[:,0]+offset
-            lat = points[:,1]
-            if(mapType == 'robin2'):
-                mask = np.logical_or(lon < 0.,lon > 360.)
-                lon[mask] = np.nan
-                lon[mask] = np.nan
-
-            (x, y) = map(lon, lat)
-                
-            mask = x == 1e30
-            x[mask] = np.nan
-            y[mask] = np.nan
-            if(filled and supportsFill):
-                xy = zip(x,y)
-                poly = Polygon( xy, facecolor=color, alpha=0.4)
-                plt.gca().add_patch(poly)
-            map.plot(x, y, linewidth=2.0, color=color)
+        (mapType, ax, projection, plotFileName, fig) = mapInfo[mapIndex]
+        x = points[:,0]
+        y = points[:,1]
+        ax.fill(x, y, transform=refProjection, color=color, alpha=0.4, zorder=3)
```

Comment:
The fill and lines should be achievable in a single call. If not, you could even use a shapely geometry to visualise.